src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Cancellation.cs

```diff
@@ -255,15 +272,11 @@ public async Task GetAsync_CancelPendingRequests_DoesntCancelReadAsyncOnResponse
 
                     Task serverTask = server.AcceptConnectionAsync(async connection =>
                     {
-                        await connection.ReadRequestHeaderAndSendCustomResponseAsync(
-                            $"HTTP/1.1 200 OK\r\n" +
-                            $"Date: {DateTimeOffset.UtcNow:R}\r\n" +
-                            $"Content-Length: {contentLength}\r\n" +
-                            $"\r\n");
-
+                                                                                                                                                                                                            await connection.ReadRequestDataAsync();
+                                                                                                                                                                                                            await connection.SendResponseAsync(HttpStatusCode.OK, headers : new HttpHeaderData[] { new HttpHeaderData("Content-Length", contentLength.ToString()) } , body : "", isFinal : false);
```

Comment:
colons

Reply:
Why is isFinal false here?

Reply:
It does not matter for HTTP1. But for Http2 we are trying to avoid sending endStream.